wordpress-mobile/WordPress-Android
# [Tooling] Update Fastfile with latest toolkit breaking changes

This is part of project ref paaHJt-2sc-p2

⚠️  This PR targets the `release/18.4` branch as I'm hoping to get it merged before this sprint's store submission in order to take the occasion to test it both for 18.4's `finalize_release` and for 18.5's `code_freeze` lanes, so that we can check nothing breaks before officially doing the major release of the toolkit.

---

## What it does

This PR points to the breaking changes introduced in https://github.com/wordpress-mobile/release-toolkit/pull/300, in preparation for the next major version of the release-toolkit.

It updates the `Fastfile` accordingly to fix the call sites of the actions that have been changed in the toolkit:

 - Removes the dummy `app: 'wp_jp'` parameters in calls for those actions.
 - Add a `get_app_name_option!` helper method which checks, for the lanes where the `app:` parameter is still needed (especially the `build_*` "internal" lanes), that the passed parameter is a valid one (either `wordpress` or `jetpack`)
 - Removes the related `# TODO` notes
 - Fix some documentation comments
 - Additional cleanup: move top-level constants like `{WP,JP}_PREFIX` and `{WP,JP}_PACKAGE_NAME` to the dedicated `APP_SPECIFIC_VALUES`

## To Test

As usual with those changes about `Fastfile`, it's super tricky to test without creating side-effects and actual releases.
Some ideas of what we could test still:

 - Run `rubocop` on the `fastlane/Fastfile`, not much to check the style violations (there are many, I actually plan to submit a subsequent PR about this on top of those changes later, but better keep those separate for review-ability), but mostly to check that there are no syntax errors and `rubocop` can read the structure of the file (i.e. there's no missing `end` for each matching `do`, etc)
 - Run `bundle exec fastlane build_internal skip_prechecks:true` and check that the lane complains that the `app:` parameter is missing
 - Run `bundle exec fastlane build_internal app:foo skip_prechecks:true` and check that the lane complains that the `app:` parameter is invalid
 - Run `bundle exec fastlane build_internal app:wordpress skip_prechecks:true` and check that the lane doesn't crash, and lints + builds wordpress Zalpha Debug
 - Run `bundle exec fastlane build_alpha app:wordpress skip_prechecks:true` and check that the lane doesn't crash, and lints + builds wordpress Zalpha Release
 - Run `bundle exec fastlane build_beta app:wordpress skip_prechecks:true` and check that the lane doesn't crash, and lints + builds wordpress Vanilla Release
